src/main/java/frc/robot/vision/sources/VisionSource.java

```diff
@@ -0,0 +1,21 @@
+package frc.robot.vision.sources;
+
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import frc.robot.vision.GyroAngleValues;
+
+import java.util.Optional;
+
+public interface VisionSource<VisionData> {
```

Comment:
```suggestion
public interface VisionSource {
```

Reply:
No, some sources will return the pose estimation of the robot and apriltags stuff while other might return an Object detection record that has the objects' pos and stuff

Reply:
So? You either do a generic interface and extend from it or don't do generazation
```suggestion
public interface VisionSource<T> {
```

Reply:
Agreed

Reply:
it's a java convention to call it T